specutils/fitting/fitmodels.py

```diff
@@ -188,29 +236,75 @@ def _fit_lines(spectrum, model, fitter=fitting.LevMarLSQFitter(),
         dispersion = dispersion[indices]
         flux = flux[indices]
 
+    elif window is not None and isinstance(window, SpectralRegion):
+        try:
+            idx1, idx2 = window.bounds
+            if idx1 == idx2:
+                raise Exception("Bad selected region.")
+            extracted_regions = extract_region(spectrum, window)
+            dispersion, flux = _combined_region_data(extracted_regions)
+            dispersion = dispersion * dispersion_unit
+            flux = flux * flux_unit
+        except ValueError as e:
+            return
+
+    if flux is None or len(flux) == 0:
+        raise Exception("Spectrum flux is empty or None.")
+
+    input_spectrum = spectrum
+
+    spectrum = Spectrum1D(flux=flux.value * flux_unit,
+                          spectral_axis=dispersion.value * dispersion_unit,
+                          wcs=input_spectrum.wcs,
+                          velocity_convention=input_spectrum.velocity_convention,
+                          rest_value=input_spectrum.rest_value)
+
     #
     # Compound models with units can not be fit.
     #
     # Convert the model initial guess to the spectral
     # units and then remove the units
     #
 
-    model_unitless, dispersion_unitless, flux_unitless = _strip_units_from_model(model, spectrum)
+    model_unitless, dispersion_unitless, flux_unitless = _strip_units_from_model(model, spectrum, convert=not ignore_units)
 
     #
     # Do the fitting of spectrum to the model.
     #
 
-    fit_model_unitless = fitter(model_unitless, dispersion_unitless, flux_unitless)
+    fit_model_unitless = fitter(model_unitless, dispersion_unitless, flux_unitless,
+                                maxiter=maxiter, acc=acc, epsilon=epsilon)
```

Comment:
in the future it is possible/likely that fitters have other keywords.  Maybe this is better as a `**kwargs` in `fit_lines` or at least a dictionary that gets passed in as `fitter(model_unitless, dispersion_unitless, flux_unitless, **fitkwargs)`?

Reply:
Agreed, it would have been really nice if astropy allowed these parameters to be defined at fitter instantiation. E.g., `fitter = LevMarLSQFitter(maxiter=200)`

Reply:
right, but I mean *now* it should still be ``kwargs`` in this specific call instead of explicitly using `maxiter`, `acc` and `epsilon`, since the fitter call's kwargs could change in the future.